src/main/java/online/carsharing/service/impl/PaymentServiceImpl.java

```diff
@@ -0,0 +1,208 @@
+                && rental.getActualReturnDate().isAfter(rental.getReturnDate())) {
+            BigDecimal overdueAmount = calculateAmount(
+                    Payment.Type.FINE, rental, rental.getCar().getDailyFee());
+            createAndSavePayment(overdueAmount, rental, Payment.Type.FINE);
+        }
+    }
+
+    @Override
+    public PaymentResponseDto createPaymentSession(PaymentRequestDto requestDto) {
+        Payment paymentEntity = paymentMapper.toEntity(requestDto);
+        Rental rental = rentalRepository.findByIdWithCar(paymentEntity.getRentalId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Rental not found with id: " + paymentEntity.getRentalId()));
+
+        if (rentalRepository.existsPendingPaymentsByUserId(rental.getUser().getId())) {
+            throw new PaymentProcessingException(
+                    "User already have active rentals or pending payments");
+        }
+
+        if (rental.getCar() == null) {
+            throw new EntityNotFoundException("Car not found with id: " + rental.getCar().getId());
+        }
+
+        BigDecimal amount = calculateAmount(
+                paymentEntity.getType(), rental, rental.getCar().getDailyFee());
+
+        Payment payment = createAndSavePayment(amount, rental, paymentEntity.getType());
+
+        return paymentMapper.toDto(payment);
+    }
+
+    @Override
+    public void markPaymentAsPaid(String sessionId) {
+        Payment payment = paymentRepository.findBySessionId(sessionId);
+        if (payment != null) {
+            payment.setStatus(Payment.Status.PAID);
+            paymentRepository.save(payment);
+        }
+    }
+
+    @Override
+    public void markPaymentAsCanceled(String sessionId) {
+        Payment payment = paymentRepository.findBySessionId(sessionId);
+        if (payment != null) {
+            payment.setStatus(Payment.Status.CANCELLED);
+            paymentRepository.save(payment);
+        }
+    }
+
+    private Session getSession(BigDecimal amount, Car car) {
+        try {
+            return Session.create(getSessionParams(amount, car));
+        } catch (StripeException e) {
+            throw new PaymentProcessingException("Stripe API exception", e);
+        }
+    }
+
+    private SessionCreateParams.LineItem.PriceData.ProductData getProductData(Car car) {
+        return SessionCreateParams.LineItem.PriceData.ProductData.builder()
+                .setName(PAYMENT_NAME
+                        + " brand: " + car.getBrand()
+                        + " model: " + car.getModel()
+                        + " type: " + car.getType())
+                .build();
+    }
+
+    private SessionCreateParams.LineItem.PriceData getPriceData(BigDecimal amount, Car car) {
+        return SessionCreateParams.LineItem.PriceData.builder()
+                .setCurrency(CURRENCY)
+                .setUnitAmount(amount.multiply(
+                        BigDecimal.valueOf(CENTS_IN_DOLLAR)).longValue())
```

Comment:
I might be wrong, but this seems a bit overcomplicated (in method). You initially set `CENTS_IN_DOLLAR` as Long, then cast it to BigDecimal, and then cast it back to Long. I believe it would be better to set `CENTS_IN_DOLLAR` as BigDecimal in the const.

Reply:
we need to talk about that